graviti/manager/common.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Common tools."""
+
+import warnings
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from graviti.manager.commit import Commit
+
+
+class StatusWarning(Warning):
+    """This class defines the warning that the commit of dataset branch or tag is not up-to-date.
+
+    Arguments:
+       revision_type: The type of current dataset revision.
+       name: The name of current dataset revision.
+
+    """
+
+    def __init__(self, revision_type: str, name: str):
+        super().__init__()
+        self._type = revision_type
+        self._name = name
+
+    def __str__(self) -> str:
+        return (
+            f"The commit id of {self._type} ({self._name}) of current dataset is not up-to-date. It"
+            f" is recommended to update the 'dataset.HEAD' by the method 'checkout('{self._name}')'"
```

Comment:
the variable name of current dataset may not be `dataset`, I think we should not use `dataset.HEAD`